core/src/sumeragi/network_topology.rs

```diff
@@ -268,6 +255,40 @@ impl Topology {
     }
 }
 
+impl<'topology> NonEmptyTopology<'topology> {
+    /// Get leader's peer id.
+    pub fn leader(&self) -> &'topology PeerId {
+        &self.topology.sorted_peers[self.topology.leader_index()]
+    }
+}
+
+impl<'topology> ConsensusTopology<'topology> {
+    /// Get proxy tail's peer id.
+    pub fn proxy_tail(&self) -> &'topology PeerId {
+        &self.topology.sorted_peers[self.topology.proxy_tail_index()]
+    }
+
+    /// Get leader's peer id.
```

Comment:
```suggestion
    /// Get leader's [`PeerId`]
```
And similarly for others.